transmission_test.go

```diff
@@ -279,92 +277,95 @@ func TestTxSendBatch(t *testing.T) {
 	}
 }
 
-func TestBatchMarshal(t *testing.T) {
-	tsts := []struct {
-		in          map[string][]map[string]interface{} // map of {dataset to list of events}
-		niledEvents map[string][]bool
-		expected    string
-		errMsgs     []string
-	}{
-		{
-			map[string][]map[string]interface{}{
-				"alpha": {
-					{"a": 1},
-					{"b": func() {}},
-					{"c": 3},
-				},
-				"beta": {
-					{"d": make(chan string)},
-					{"e": 5},
-				},
-				"gamma": {
-					{"f": 6},
-				},
-			},
-			map[string][]bool{
-				"alpha": {false, true, false},
-				"beta":  {true, false},
-				"gamma": {false},
-			},
-			`{"alpha":[{"data":{"a":1}},{"data":{"c":3}}],"beta":[{"data":{"e":5}}],"gamma":[{"data":{"f":6}}]}`,
-			[]string{"MarshalJSON", "MarshalJSON"},
-		},
-		{ // shouldn't happen based on how muster works, but just in case
-			map[string][]map[string]interface{}{"nada": {}},
-			map[string][]bool{"nada": []bool{}},
-			"{}",
-			nil,
-		},
-		{
-			map[string][]map[string]interface{}{
-				"nada": {
-					{"a": func() {}},
-					{"b": make(chan string)},
-				},
-				"zip": {
-					{"c": func() {}},
-				},
-			},
-			map[string][]bool{
-				"nada": {true, true},
-				"zip":  {true},
-			},
-			"{}",
-			[]string{"MarshalJSON", "MarshalJSON", "MarshalJSON"},
-		},
-	}
+// TODO figure out how to test multiple datasets, writekeys, and apiHosts
+// causing multiple batches to be dispatched
```

Comment:
suggested test events:

```
Event{WriteKey: "w1", Dataset: "alpha", APIHost: "https://api.hny.co"},
Event{WriteKey: "w1", Dataset: "beta", APIHost: "https://api.hny.co"},
Event{WriteKey: "w2", Dataset: "alpha", APIHost: "https://api.hny.co"},
Event{WriteKey: "w1", Dataset: "alpha", APIHost: "https://ui.hny.co"}
```

and it seems like you'll want to use a `FakeRoundTripper` + map of expected payloads.